packages/grid/_modules_/grid/components/toolbar/DensitySelector.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import MenuItem from '@material-ui/core/MenuItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import { useIcons } from '../../hooks/utils/useIcons';
+import { ApiContext } from '../api-context';
+import { DensityTypes, Density } from '../../models/gridOptions';
+import { useGridSelector } from '../../hooks/features/core/useGridSelector';
+import { optionsSelector } from '../../hooks/utils/useOptionsProp';
+import { DensityOption } from '../../models/api/densityApi';
+import { densityValueSelector } from '../../hooks/features/density';
+import { GridMenu } from '../menu/GridMenu';
+
+const DensitySelectorListItemText = withStyles({
+  root: {
+    textTransform: 'capitalize',
+  },
+})(ListItemText);
+
+export function DensitySelector() {
+  const apiRef = React.useContext(ApiContext);
+  const { density, rowHeight, headerHeight } = useGridSelector(apiRef, optionsSelector);
+  const densityValue = useGridSelector(apiRef, densityValueSelector);
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const icons = useIcons();
+
+  const DensityCompactIcon = icons!.DensityCompact!;
+  const DensityStandardIcon = icons!.DensityStandard!;
+  const DensityComfortableIcon = icons!.DensityComfortable!;
+
+  const DensityOptions: Array<DensityOption> = [
+    {
+      icon: <DensityCompactIcon />,
+      label: DensityTypes.Compact,
+    },
+    {
+      icon: <DensityStandardIcon />,
+      label: DensityTypes.Standard,
+    },
+    {
+      icon: <DensityComfortableIcon />,
+      label: DensityTypes.Comfortable,
+    },
+  ];
```

Comment:
It will need to be translated, I don't think that we can use the code constants for it.

```suggestion
  const DensityOptions: Array<DensityOption> = [
    {
      icon: <DensityCompactIcon />,
      label: 'Compact',
    },
    {
      icon: <DensityStandardIcon />,
      label: 'Standard',
    },
    {
      icon: <DensityComfortableIcon />,
      label: 'Comfortable',
    },
  ];
```

Reply:
Updated